components/org.wso2.carbon.identity.application.authentication.handler.identifier/src/main/java/org/wso2/carbon/identity/application/authentication/handler/identifier/UserResolveExecutor.java

```
@@ -0,0 +1,163 @@
+                String primaryDomain = currentUSM.getRealmConfiguration()
+                        .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME);
+
+                while (currentUSM != null) {
+                    String domain = currentUSM.getRealmConfiguration()
+                            .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME);
+
+                    if (StringUtils.isNotBlank(domain) && !domain.equalsIgnoreCase(primaryDomain)) {
+                        String qualifiedUsername = domain + UserCoreConstants.DOMAIN_SEPARATOR + username;
+                        if (userStoreManager.isExistingUser(qualifiedUsername)) {
+                            resolvedUsername = qualifiedUsername;
+                            retrievedClaims = userStoreManager
+                                    .getUserClaimValues(resolvedUsername, claimsToFetch, null);
+                            break;
+                        }
+                    }
+                    currentUSM = currentUSM.getSecondaryUserStoreManager();
+                }
+            }
+
+            if (retrievedClaims != null && !retrievedClaims.isEmpty()) {
+                context.getFlowUser().addClaim(CLAIM_URI_EMAIL, retrievedClaims.get(CLAIM_URI_EMAIL));
+                context.getFlowUser().addClaim(CLAIM_URI_FIRST_NAME, retrievedClaims.get(CLAIM_URI_FIRST_NAME));
+                context.getFlowUser().addClaim(CLAIM_URI_LAST_NAME, retrievedClaims.get(CLAIM_URI_LAST_NAME));
+                context.getFlowUser().addClaim(CLAIM_URI_MOBILE, retrievedClaims.get(CLAIM_URI_MOBILE));
+            }
+
+        } catch (UserStoreException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error while fetching attributes for : " +
+                        LoggerUtils.getMaskedContent(username) + " in tenant: " + tenantDomain, e);
+            }
+        } catch (Exception e) {
```

We need to propagate the error to the engine